### PR TITLE
[web] New filter options for CodeChecker cmd runs command

### PR DESCRIPTION
**Pull request by csordasmarton**
* Extend Thrift API to be able to filter runs on multiple conditions.
* Add the same filter command line options to the `CodeChecker cmd runs`
  as the `CodeChecker cmd del` command.
* Add new test cases.

### CODE REVIEW

File: docs/web/user_guide.md [hunk 1]
```diff
@@ -757,6 +757,24 @@ optional arguments:
...
+  --all-before-run RUN_NAME
+                        Get all runs that were stored to the server BEFORE the
+                        specified one.
+  --all-after-run RUN_NAME
+                        Get all runs that were stored to the server AFTER the
+                        specified one.
+  --all-after-time TIMESTAMP
+                        Get all analysis runs that were stored to the server
+                        AFTER the given timestamp. The format of TIMESTAMP is
+                        'year:month:day:hour:minute:second' (the "time" part
```
[the remark this answers is not in the document]
Comment (csordasmarton) [reply docs/web/user_guide.md]: @gyorb Do we want to use different command line options for `CodeChecker cmd del` and `CodeChecker cmd runs`. So I can not use the **CodeChecker cmd runs** command easily to see which runs will be removed if I use the **CodeChecker cmd del** command.

File: web/api/v6/report_server.thrift [hunk 2]
```diff
@@ -277,9 +277,13 @@ typedef list<CommentData> CommentDataList
  * If exactMatch field is True it will use exact match for run names.
  */
 struct RunFilter {
-  1: list<i64>    ids,        // IDs of the runs.
-  2: list<string> names,      // Part of the run name.
-  3: bool         exactMatch  // If it's True it will use an exact match for run names.
+  1: list<i64>       ids,        // IDs of the runs.
+  2: list<string>    names,      // Part of the run name.
+  3: bool            exactMatch, // If it's True it will use an exact match for run names.
```
[the remark this answers is not in the document]
Comment (csordasmarton) [reply web/api/v6/report_server.thrift]: Assume we have multiple runs: `runname1`, `runname2`, `runname*`.

If we filter on `runname*` and the `exactMatch` if `False` we will replace the `*` quantifier with `%` and the results will be all the runs.
https://github.com/Ericsson/codechecker/blob/e0f601f13d6d6cea51c231612b755be56af24a4f/web/server/codechecker_server/api/report_server.py#L324-L330

If we filter on `runname*` and the `exactMatch` if `True` the results will be only the `runname*`.

So we introduced this to be able to filter runs which contain `*` character in there names.

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- web/tests/functional/delete_runs/test_delete_runs.py  6
         

Clones added
============
- web/tests/functional/delete_runs/test_delete_runs.py  2
- web/tests/functional/detection_status/test_detection_status.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/codechecker/codechecker/pullRequest?prid=4231395&bid=14464578)

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- web/tests/functional/delete_runs/test_delete_runs.py  6
         

Clones added
============
- web/tests/functional/detection_status/test_detection_status.py  2
- web/tests/functional/delete_runs/test_delete_runs.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeChecker/codechecker/pullRequest?prid=4231395&bid=14464578)

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- web/tests/functional/delete_runs/test_delete_runs.py  6
         

Clones added
============
- web/tests/functional/detection_status/test_detection_status.py  2
- web/tests/functional/delete_runs/test_delete_runs.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeChecker/codechecker/pullRequest?prid=4231395&bid=14464578)

**Pull request merged**